setup.py

```diff
@@ -79,7 +78,7 @@ def find_version(*file_paths):
         # here.
         # Restricted version because python-for-android's recipes can't handle
         # later versions.
-        'cython<3.0'
+        'cython<3.12'
```

Comment:
Are we sure we want to pull some of these changes in this pull request?
Some may deserve its own pull request to discuss it and to make it easier to spot regression and potentially revert